docker/Dockerfile.intel

```diff
@@ -0,0 +1,90 @@
+##############################################################
+# This Dockerfile contains Devito codes + Intel HPC toolkit.
+#
+# BUILD:
+#   docker build --network=host --file docker/Dockerfile.intel --tag devito/intel:1.0 .
+#
+# RUN:
+#   docker run --rm -it devito/intel:1.0
+#
+# Start a terminal session:
+#   docker run --rm -it devito/intel:1.0 /bin/bash
+#
+# RUN a few tests:
+#   docker run --rm --name testrun 'devito/intel:1.0' python -m pytest -m "not parallel" tests/test_operator.py
+#
+##############################################################
+# Inherited from: https://github.com/intel/oneapi-containers/blob/master/images/docker/hpckit/Dockerfile.ubuntu-18.04
+# Copyright (c) 2019-2020 Intel Corporation.
+# SPDX-License-Identifier: BSD-3-Clause
+
+FROM intel/oneapi-hpckit:latest
```

Comment:
This and language are the oly difference with the normal dockerfile. Would be better to be merged and handled via cmd args. i.e first line having
```
ARG BASE=python:3.7
FROM $BASE
```

and call it with `build-args:` in the CI file. (same for DEVITO_ARCH)

Reply:
ah, lol, I just wrote about this code duplication issue a few lines above